packages/babel-plugin-transform-react-jsx/src/create-plugin.js

```diff
@@ -411,9 +411,17 @@ You can set \`throwIfNamespace: false\` to bypass this warning.`,
             case "__self":
               if (extracted[name]) throw sourceSelfError(path, name);
             /* falls through */
-            case "key":
-              extracted[name] = convertAttributeValue(attr.node.value);
+            case "key": {
+              const keyValue = convertAttributeValue(attr.node.value);
+              if (keyValue === null) {
+                throw path.buildCodeFrameError(
+                  'Provide an explicit value to be used as a key. You are not supposed to use the "key" shorthand for "key={true}".',
```

Comment:
I'm not a native speaker so I might just be reading this wrong, but this message _feels_ a bit aggressive to me.

Maybe this?

```suggestion
                  'Please provide an explicit key value. Using "key" as a shorthand for "key={true}" is not allowed.',
```

Reply:
Agreed

Reply:
Okay, this seems nicer. I've included the change in a seperate commit.